src/psyclone/psyir/transformations/inline_trans.py

```diff
@@ -473,38 +472,91 @@ def validate(self, node, options=None):
 
     @staticmethod
     def _find_routine(call_node):
-        '''
-        Searches for the definition of the routine that is being called by
+        '''Searches for the definition of the routine that is being called by
         the supplied Call.
 
-        Currently only supports routines that are present in the
-        same source file - TODO #924.
-
         :param call_node: the Call that is to be inlined.
         :type call_node: :py:class:`psyclone.psyir.nodes.Call`
 
         :returns: the PSyIR for the target routine.
         :rtype: :py:class:`psyclone.psyir.nodes.Routine`
 
-        :raises TransformationError: if the RoutineSymbol is not local.
-        :raises TransformationError: if the routine symbol is local but the \
-            definition cannot be found.
+        :raises TransformationError: if the routine definition cannot be found.
+
         '''
         name = call_node.routine.name
         routine_sym = call_node.scope.symbol_table.lookup(name)
-        if not routine_sym.is_local:
-            raise TransformationError(
-                f"Routine '{name}' is imported and therefore cannot currently "
-                f"be inlined - TODO #924.")
-        table = routine_sym.find_symbol_table(call_node)
-        for routine in table.node.walk(Routine):
-            if routine.name == name:
+
+        if routine_sym.is_local:
+            table = routine_sym.find_symbol_table(call_node)
+            for routine in table.node.walk(Routine):
+                if routine.name == name:
+                    return routine
+
+        elif routine_sym.is_unresolved:
+
+            # First check for any wildcard imports and see if they can
+            # be used to resolve the symbol.
+            current_table = call_node.scope.symbol_table
+            while current_table:
+                for container_symbol in current_table.containersymbols:
+                    if container_symbol.wildcard_import:
+                        routine = InlineTrans._find_routine_in_container(
+                            call_node, container_symbol)
+                        if routine:
+                            return routine
+                current_table = current_table.parent_symbol_table()
+
+            # Next check for any "raw" Routines, i.e. ones that are not
+            # in a Container.  Such Routines would exist in the PSyIR
+            # as a child of a FileContainer (if the PSyIR contains a
+            # FileContainer). Note, if the PSyIR does contain a
+            # FileContainer, it will be the root node of the PSyIR.
+            for routine in call_node.root.children:
+                if isinstance(routine, Routine) and routine.name == name:
+                    return routine
+
```

Comment:
OK.

Reply:
Thanks. Just a tweak to the error message: "...imports from {wildcard_names} and all..." - i.e. add "from" and remove the quotes since `wildcard_names`  is a list.

Reply:
Updated.